components/wso2is.key.manager.tokenpersistence/src/main/java/org/wso2/is/key/manager/tokenpersistence/dao/DBInvalidTokenPersistence.java

```diff
@@ -0,0 +1,400 @@
+                        IdentityDatabaseUtil.commitTransaction(connection);
+                        if (retryAttemptCounter > 0) {
+                            log.info("Successfully recovered CON_SUB_EVT_KEY constraint violation with the attempt : "
+                                    + retryAttemptCounter);
+                        }
+                    } catch (SQLIntegrityConstraintViolationException e) {
+                        log.warn("User event token revocation rule already persisted.");
+                        IdentityDatabaseUtil.rollbackTransaction(connection);
+                        if (retryAttemptCounter >= getTokenPersistRetryCount()) {
+                            log.error("CON_SUB_EVT_KEY constraint violation retry count exceeds above the maximum "
+                                    + "count - " + getTokenPersistRetryCount());
+                            throw new IdentityOAuth2Exception(errorMessage, e);
+                        }
+                        revokeTokensByUserEvent(subjectId, subjectIdType, revocationTime, organization,
+                                retryAttemptCounter + 1);
+                    } catch (SQLException e) {
+                        log.warn("User event token revocation rule already persisted.");
+                        IdentityDatabaseUtil.rollbackTransaction(connection);
+                        // Handle constrain violation issue in JDBC drivers which does not throw
+                        // SQLIntegrityConstraintViolationException
+                        if (StringUtils.containsIgnoreCase(e.getMessage(), "CON_SUB_EVT_KEY")) {
+                            if (retryAttemptCounter >= getTokenPersistRetryCount()) {
+                                log.error("'CON_SUB_EVT_KEY' constrain violation retry count exceeds above the maximum "
+                                        + "count - " + getTokenPersistRetryCount());
+                                throw new IdentityOAuth2Exception(errorMessage, e);
+                            }
+                            revokeTokensByUserEvent(subjectId, subjectIdType, revocationTime, organization,
+                                    retryAttemptCounter + 1);
+                        } else {
+                            throw new IdentityOAuth2Exception("Error while inserting user event revocation rule to db."
+                                    + e.getMessage(), e);
+                        }
+                    } catch (Exception e) {
+                        log.warn("User event token revocation rule already persisted.");
+                        IdentityDatabaseUtil.rollbackTransaction(connection);
+                        // Handle constrain violation issue in JDBC drivers which does not throw
+                        // SQLIntegrityConstraintViolationException or SQLException.
+                        if (StringUtils.containsIgnoreCase(e.getMessage(), "CON_SUB_EVT_KEY") || (e.getCause() != null
```

Comment:
Shall we try to unify these two catch blocks since the content is almost the same? Maybe using instanceof as an elseif block?